testing/integration/kubernetes_agent_standalone_test.go

```diff
@@ -1118,3 +934,233 @@ func k8sGetContext(t *testing.T, info *define.Info) k8sContext {
 		enrollParams:   enrollParams,
 	}
 }
+
+// k8sTestStep is a function that performs a single step in a k8s integration test
+type k8sTestStep func(t *testing.T, ctx context.Context, kCtx k8sContext, namespace string)
+
+// k8sStepCreateNamespace creates a namespace for the current test and adds a test cleanup that
+// deletes it
+func k8sStepCreateNamespace() k8sTestStep {
+	return func(t *testing.T, ctx context.Context, kCtx k8sContext, namespace string) {
+		k8sNamespace := &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: namespace,
+			},
+		}
+
+		t.Cleanup(func() {
+			err := k8sDeleteObjects(ctx, kCtx.client, k8sDeleteOpts{wait: true}, k8sNamespace)
```

Comment:
Maybe worth skipping finalizers here? Deleting namespaces can sometimes take a long time.

Reply:
I added this cleanup to accommodate running the kubernetes integration tests locally. In CI our env is provisioned at every CI run. But locally the k8s cluster might and most probably be already there. Thus if we don't cleanup the namespace we are gonna get an error at the creation of the next run. We could always ignore error exists but I don't consider that the best idea. Also because the namespace here is one that only the test deploys the corresponding k8s objects which are cleaned up in other steps, it's not gonna be that slow most of the times it is empty. And as an extra delay minimisation here the default timeout for waiting to delete k8s object is just 20 secs. Lastly we never fail a test because of an error at delete. Thus I would propose to leave this delete. Does it make sense? 🙂

Reply:
> then making the namespace unique per test run is better imo.

It is already unique across the different tests but if you rerun the same test it will get the same namespace (it derives from the test name). And now that I think about it again, we do need that cleanup for our CI as well, because we test different docker variants on the same k8s cluster. That said, I share similar experiences to you, but whenever I resorted to force delete a namespace it was never faster than normal delete and the reason behind a namespace being slow at delete, is weird finalisers at the objects contained in this namespace, which we don't have. So coming back to my original opinion I believe this delete should stay here 🙂

Reply:
Allright, how about attaching a timestamp to the namespace name then? Or even https://github.com/dustinkirkland/golang-petname if we want to be fancy. Would be nice if `testing.T` exposed a run id, but unfortunately no such luck.

Reply:
I see no problem in cleaning up the test environment when the test is done, unless we have metrics showing that this is heavily impacting our test run time